.github/workflows/lint_build_test.yaml

```diff
@@ -0,0 +1,164 @@
+name: Lint and Build
+
+on:
+  pull_request:
+
+env:
+  NODE_VERSION: 10
+  GO_VERSION: 1.15
+  JAVA_VERSION: 9
+
+jobs:
+  javascript:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: JavaScript - kyber
+        run: |
+          cd external/js/kyber
+          npm ci
+          npm run linter
+          npm run build
+
+      - name: JavaScript - cothority
+        run: |
+          cd external/js/cothority
+          npm ci
+          npm run linter
+          npm link @dedis/kyber
+          npm run build
+
+  golang:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        golang-version: [1.14, 1.15] # Cannot use env. here :(
+
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.golang-version }}
+
+      - name: Installing golint
+        run: GO111MODULE=off go get -u golang.org/x/lint/golint
+
+      - name: Testing proto-files
+        run: |
+          make test_proto
+          go version
+
+      - name: Running make-tests
+        run: |
+          make -C conode verify
+          make test_{fmt,lint}
+
+  # As pre-commit doesn't like to be run in parallel, this step runs it only
+  # for the latest go-version.
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-python@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Installing golint
+        run: GO111MODULE=off go get -u golang.org/x/lint/golint
+
+      - name: Running pre-commit
+        uses: pre-commit/action@v2.0.0
+
+  # The barrier job is used to avoid writing a lot of needs
+  barrier:
+    runs-on: ubuntu-latest
+    needs:
+      - javascript
+      - golang
+      - pre-commit
+    steps:
+      - name: Done linting and building
+        run: echo Linting and Building done
```

Comment:
Do you really need a `steps` block? It would be nice to avoid spawning a VM for a simple echo.

Reply:
yep, needed: 

![Screenshot 2020-12-08 at 14 42 22](https://user-images.githubusercontent.com/1545067/101491130-a6ec9480-3963-11eb-9916-faf383a240a4.png)